### PR TITLE
Mpsl with api docs

### PR DESCRIPTION
**Pull request by rugeGerritsen**
Also includes some ble controller size and stack usage improvements.

Manifest update: https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/2045

### CODE REVIEW

File: mpsl/include/mpsl_temp.h [hunk 1]
```diff
@@ -5,11 +5,13 @@
...
- * @{
+ * @file mpsl_temp.h
  *
- * @brief APIs for the MPSL Temp module.
+ * @defgroup mpsl_temp MPSL Temperature Sensor interface
+ * @ingroup  mpsl
  *
+ * The MPSL temperature sensor interface provides APIs for measuring the chip temperature.
+ * @{
  */
```
Comment (b-gent) [inline mpsl/include/mpsl_temp.h]: in line 37 in this file, @retval should be @return

File: mpsl/doc/api.rst [hunk 2]
```diff
@@ -0,0 +1,46 @@
+.. mpsl_api:
+
+API documentation
+#################
+
+MPSL
```
Comment (b-gent) [inline mpsl/doc/api.rst]: This could be a bit more specific, like 'MPSL API interface'

**Comment by b-gent** [comment]
Only small suggestions for improvement.

**Comment by rugeGerritsen** [comment]
> Only small suggestions for improvement.

Thanks. I've updated it now

**Pull request merged**